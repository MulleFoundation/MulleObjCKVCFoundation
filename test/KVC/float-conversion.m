//
//  main.m
//  archiver-test
//
//  Created by Nat! on 19.04.16.
//  Copyright Â© 2016 Mulle kybernetiK. All rights reserved.
//


#import <MulleObjCKVCFoundation/MulleObjCKVCFoundation.h>


@interface Foo : NSObject

@property( assign) float f;
@property( assign) double d;
@property( assign) long double ld;

@end


@implementation Foo

- (void) dump
{
   fprintf( stdout, "%%f\n");
   fprintf( stdout, "\tf   = %f\n", _f);
   fprintf( stdout, "\td   = %f\n", _d);
   fprintf( stdout, "\tld  = %Lf\n", _ld);

   fprintf( stdout, "%%g\n");
   fprintf( stdout, "\tf   = %g\n", _f);
   fprintf( stdout, "\td   = %g\n", _d);
   fprintf( stdout, "\tld  = %Lg\n", _ld);

   fprintf( stdout, "Mulle\n");
   fprintf( stdout, "\tf   = %0.8g\n", _f);
   fprintf( stdout, "\td   = %0.17g\n", _d);
   fprintf( stdout, "\tld  = %0.21Lg\n", _ld);

   fprintf( stdout, "Apple (does no ld)\n");
   fprintf( stdout, "\tf   = %0.7g\n", _f);
   fprintf( stdout, "\td   = %0.16g\n", _d);
   fprintf( stdout, "\tld  = %0.21Lg\n", _ld);

   // Voodoo check
   fprintf( stdout, "Mulle Accessor\n");
   fprintf( stdout, "\tf   = %0.8g\n", [self f]);
   fprintf( stdout, "\td   = %0.17g\n", [self d]);
   fprintf( stdout, "\tld  = %0.21Lg\n", [self ld]);

   // Voodoo check 2
   fprintf( stdout, "Mulle KVC Accessor\n");
   fprintf( stdout, "\tf   = %0.8g\n", [[self valueForKey:@"f"] floatValue]);
   fprintf( stdout, "\td   = %0.17g\n", [[self valueForKey:@"d"] doubleValue]);
   fprintf( stdout, "\tld  = %0.21Lg\n", [[self valueForKey:@"ld"] longDoubleValue]);
}

@end


// just a little check written when a float got printed ugly...

int  main( int argc, const char * argv[])
{
   NSNumber   *nr;
   Foo        *foo;

   nr  = [NSNumber numberWithDouble:18.48];

   foo = [Foo object];
   [foo setValue:nr
          forKey:@"f"];
   [foo setValue:nr
          forKey:@"d"];
   [foo setValue:nr
          forKey:@"ld"];

   [foo dump];

   return( 0);
}
