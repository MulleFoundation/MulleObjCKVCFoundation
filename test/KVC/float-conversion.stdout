%f
	f   = 18.480000
	d   = 18.480000
	ld  = 18.480000
%g
	f   = 18.48
	d   = 18.48
	ld  = 18.48
Mulle
	f   = 18.48
	d   = 18.48
	ld  = 18.4800000000000004263
Apple (does no ld)
	f   = 18.48
	d   = 18.48
	ld  = 18.4800000000000004263
Mulle Accessor
	f   = 18.48
	d   = 18.48
	ld  = 18.4800000000000004263
Mulle KVC Accessor
	f   = 18.48
	d   = 18.48
	ld  = 18.4800000000000004263
